javascript-carousel/main.js

```diff
@@ -0,0 +1,84 @@
+const leftArrow = document.querySelector('.left-arrow');
+const rightArrow = document.querySelector('.right-arrow');
+const progressDot = document.querySelectorAll('.circle');
+const circleIcon = document.querySelectorAll('.circle-icon');
+const carouselImg = document.querySelector('.carousel-img');
+let currentIndex = 0;
+let intervalId;
+
+const images = [
+  {
+    src: '001.png',
+  },
+  {
+    src: '004.png',
+  },
+  {
+    src: '007.png',
+  },
+  {
+    src: '025.png',
+  },
+  {
+    src: '039.png',
+  },
+];
+
+function updateImage() {
+  carouselImg.setAttribute('src', `./images/${images[currentIndex].src}`);
+  updateProgressDots(currentIndex);
+}
+
+function resetInterval() {
+  clearInterval(intervalId);
+  intervalId = setInterval(() => {
+    currentIndex++;
+    if (currentIndex >= images.length) {
+      currentIndex = 0;
+    }
+    updateImage();
+  }, 3000);
+}
+
+resetInterval();
```

Comment:
I would move to the bottom of the file to show that this is ran "at the start" since JS is read top to bottom